### PR TITLE
[13.0][IMP] maintenance_timesheet: Add timesheet page from maintenance request form view.

### PR DESCRIPTION
**Pull request by victoralmau**
Add timesheet page from maintenance request form view.

ejemplo

Please @pedrobaeza and @ernestotejeda can you review it?

@Tecnativa TT40881

### CODE REVIEW

**Review by pedrobaeza — APPROVED** [review]

File: base_maintenance/views/maintenance_request_views.xml [hunk 1]
```diff
@@ -12,6 +12,13 @@
                     <!--Empty, to be inherited by other modules.-->
                 </div>
             </xpath>
+            <field name="description" position="replace">
```
Comment (dalonsod) [inline base_maintenance/views/maintenance_request_views.xml:16]: With a `replace` in a view, a lower priority for the view is recommended

**Review by dalonsod — CHANGES_REQUESTED** [review]
See comment.

And also,
- Totally agree making `timesheet_total_hours` stored, kanban and tree views will have a better performance. I'm not sure but, a migration hook should be neeeded in this case for older requests and timesheets?
- Timesheets original button box for a request is still present, in fact, I use and like it. Anyway, I don't know if this redundancy is the right approach.
- `base_maintenance` is also changed. A standalone PR or, at least, an individual commit, should be made instead?

File: base_maintenance/views/maintenance_request_views.xml (hunk 1 above)
Comment (victoralmau) [reply base_maintenance/views/maintenance_request_views.xml:16]: Ok, already changed in a separate commit

**Comment by victoralmau** [comment]
>     * Totally agree making `timesheet_total_hours` stored, kanban and tree views will have a better performance. I'm not sure but, a migration hook should be neeeded in this case for older requests and timesheets?
It is not necessary, when updating the module, the compute will define the value.
>     * Timesheets original button box for a request is still present, in fact, I use and like it. Anyway, I don't know if this redundancy is the right approach.
Exactly, I didn't want to remove the smart-button in case someone uses it.
>     * `base_maintenance` is also changed. A standalone PR or, at least, an individual commit, should be made instead?
Ok, I have separated it in another commit.

**Comment by dalonsod** [comment]
> It is not necessary, when updating the module, the compute will define the value.

OK, anyway IMO a good option in this case should be a pre-migration SQL script in order to prevent a heavy amount of maintenance requests. What do you think? cc @pedrobaeza 

Tested in v12, the update query could be as follows:

```
UPDATE maintenance_request
SET timesheet_total_hours=coalesce(tt.total_unit_amount, 0.0)
FROM (
    SELECT mr.id, SUM(aal.unit_amount) total_unit_amount
    FROM maintenance_request mr
    LEFT JOIN account_analytic_line aal ON aal.maintenance_request_id = mr.id
    GROUP BY mr.id
) AS tt
WHERE maintenance_request.id = tt.id
```

**Comment by pedrobaeza** [comment]
I don't think the upgrade time is going to be very huge, but OK for the migration script.

**Comment by victoralmau** [comment]
Migration script added.

**Comment by victoralmau** [comment]
With these changes merged, I will add in `maintenance_timesheet_time_control` the buttons in this tree, or better to do it already here?

**Review by dalonsod — APPROVED** [review]
👍

**Pull request merged**